src/aiida/tools/archive/create.py

```diff
@@ -280,94 +288,130 @@ def querybuilder():
 
     EXPORT_LOGGER.report(f'Creating archive with:\n{tabulate(count_summary)}')
 
+    # Handle temporary directory configuration
+    tmp_prefix = '.aiida-export-'
+    if tmp_dir is not None:
+        tmp_dir = Path(tmp_dir)
+        if not tmp_dir.exists():
+            EXPORT_LOGGER.warning(f"Specified temporary directory '{tmp_dir}' doesn't exist. Creating it.")
```

Comment:
This behaviour is different from the CLI command, that _does_ expect the temporary directory to exist. It also seems to conflict with the docstring of the function.